moving a temporary object prevents copy elision

Building the SDL/scons version on macOS 12.7.2 with clang 17.0.6 (and with `-Wextra`, if that's relevant) I get several warnings but the most frequent (occurring 110 times due to the many places this file is included) is:

```
Concurrency/AsyncTaskQueue.hpp:123:14: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
  123 |                         thread_ = std::move(std::thread{
      |                                   ^
Concurrency/AsyncTaskQueue.hpp:123:14: note: remove std::move call here
  123 |                         thread_ = std::move(std::thread{
      |                                   ^~~~~~~~~~
1 warning generated.
```

--- a/Concurrency/AsyncTaskQueue.hpp
+++ b/Concurrency/AsyncTaskQueue.hpp
@@ -120,7 +120,7 @@ template <bool perform_automatically, bool start_immediately = true, typename Pe
 		///
 		/// This is not guaranteed safely to restart a stopped queue.
 		void start() {
-			thread_ = std::move(std::thread{
+			thread_ = std::thread{
 				[this] {
 					ActionVector actions;
 
@@ -144,7 +144,7 @@ template <bool perform_automatically, bool start_immediately = true, typename Pe
 						actions.clear();
 					}
 				}
-			});
+			};
 		}
 
 		/// Schedules any remaining unscheduled work, then blocks synchronously

